packages/docusaurus-plugin-google-gtag/src/utils.ts

```diff
@@ -0,0 +1,8 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+export {useGoogleAnalytics} from './utils/useGoogleAnalytics';
```

Comment:
I still don't understand why we need this abstraction in the first place. 

The previous conversation is not resolved: if you can't explain clearly why we need this, we won't add it 😅 

I believe it ads no value

Reply:
This allows developers to send custom events to GA without having to go through the `gtag.configure` call all over again.

We have added some functionality to our DS sites that does not result in a navigation event (basically it displays or hides some content on the page based off a click event). We need to be able to track those clicks. To do this, we need to send [custom events](https://developers.google.com/analytics/devguides/collection/gtagjs/events) to GA.

With this hook the code needed to send the custom events is much cleaner and easier, we don't have to copy/paste the `configure` boilerplate in individual React components, its type-safe and does not require `ts-ignore` statements for the `window.gtag()` call, nor do we need to create an internal service to reduce the boilerplate. Or, more specifically, I decided instead of writing that service for only our internal use I figured I'd give back to the community. 

Also, its best to avoid re-calling `configure` because every time the `configure` call is made a new pageview event is generated unless its disabled as documented [here](https://developers.google.com/analytics/devguides/collection/gtagjs/pages#manual_pageviews). Without disabling the default behavior pageview events will be doubled in count. While easy to disable, its also very easy to forget to disable it (ask me how I know that 😅).

Reply:
I don't understand what you mean by "configure", I only see this in the doc: `gtag("config",...)`

This is already handle in the plugin:

```js
          {
            tagName: 'script',
            innerHTML: `
              window.dataLayer = window.dataLayer || [];
              function gtag(){dataLayer.push(arguments);}
              gtag('js', new Date());
              gtag('config', '${trackingID}', { ${
              anonymizeIP ? "'anonymize_ip': true" : ''
            } });`,
```

There's no reason your code should call this "config" action more than once IMHO, or you'd have to explain to me what you want to achieve exactly by calling configure multiple times. It's already done for you, and you can forget about it.

Just write this in your code:

```jsx
<button onClick={e => {
	  gtag('event', <action>, {
	  'event_category': <category>,
	  'event_label': <label>,
	  'value': <value>
	});
}}/>

```

No nook needed